instrumentation/jms-1.1/src/main/java/io/opentelemetry/auto/instrumentation/jms/JMSDecorator.java

```diff
@@ -80,4 +83,35 @@ public static String toSpanName(Destination destination) {
     }
     return "destination";
   }
+
+  public void afterStart(Span span, String spanName, Message message) {
+    if (spanName.startsWith("queue/")) {
+      SemanticAttributes.MESSAGING_DESTINATION_KIND.set(span, "queue");
+      SemanticAttributes.MESSAGING_DESTINATION.set(span, spanName.replaceFirst("queue/", ""));
+    } else if (spanName.startsWith("topic/")) {
+      SemanticAttributes.MESSAGING_DESTINATION_KIND.set(span, "topic");
+      SemanticAttributes.MESSAGING_DESTINATION.set(span, spanName.replaceFirst("topic/", ""));
+    }
+    if (spanName.equals("queue/<temporary>") || spanName.equals("topic/<temporary>")) {
+      SemanticAttributes.MESSAGING_TEMP_DESTINATION.set(span, true);
+    }
```

Comment:
when we move to the new `Tracer` design, and do both `JMSDecorator.toSpanName()` and `afterStart()` inside of `startSpan`, we can consolidate the conditionals between these two methods (and remove the `startsWith` logic)

Reply:
yeah